src/Components/App/CompanySearchPage.tsx

```diff
@@ -94,55 +107,105 @@ const Results = connectStateResults(
+                                            : selectedCompanies.filter((s) => s !== c.slug)
+                                    )
+                                }
+                                id={`company-pack-choose-${c.slug}`}
+                            />
+                            <label for={`company-pack-choose-${c.slug}`}>{c.name}</label>
+                        </li>
+                    );
+                })}
+            </ul>
+        </>,
+        {
+            positiveText: t(
+                'add-n-companies',
+                'generator',
+                { n: `${selectedCompanies.length}` },
+                selectedCompanies.length
+            ),
+            onPositiveFeedback: () =>
+                Promise.all(
+                    selectedCompanies
+                        .filter((slug) => !Object.keys(batch || {}).includes(slug))
+                        .map((slug) => appendToBatchBySlug(slug))
+                ).then(hideModal),
+        }
+    );
 
     return (
-        <section className="company-suggestion-pack">
-            <h1>{props.title}</h1>
-            <ul>
-                {
-                    props.companies.map((item, i) => (
-                        <li className={i > 3 && !showAllCompanies ? 'sr-only' : ''}>{item}</li>
-                    )) /* We only need to hide the other items for visual clarity, screen readers can easily skip it. See: https://ux.stackexchange.com/a/127486*/
-                }
-                {props.companies.length > 4 && (
-                    <li aria-hidden={true}>
-                        <button className="button-unstyled" onClick={() => setShowAllCompanies(!showAllCompanies)}>
-                            Show more…
-                        </button>
-                    </li>
+        <IntlProvider scope="company-packs" definition={window.I18N_DEFINITION}>
+            {/* TODO: This breaks the flex layout. -.- */}
+            <Modal />
+
+            <section className="company-suggestion-pack">
+                <h1>{t(`${pack.slug as 'address-brokers'}-title`, 'company-packs')}</h1>
+                {pack.type === 'add-all' && (
+                    <span className="icon icon-star color-yellow-600" title={t('add-all-star', 'company-packs')} />
                 )}
-            </ul>
-            <button className="company-suggestion-pack-cta button">Add to your requests</button>
-        </section>
+                <p>
+                    {pack.companies
+                        .slice(0, 7)
+                        .map((c) => c.name)
+                        .join(', ')}
+                    {pack.companies.length > 7 && (
+                        <em>
+                            <Text id="and-n-more" fields={{ n: pack.companies.length - 7 }} />
+                        </em>
+                    )}
+                </p>
+                <button className="company-suggestion-pack-cta button" onClick={showModal}>
```

Comment:
I think we need to indicate somehow if (some of) a company pack was added. Currently, the user can not be sure just from the company packs UI.